### PR TITLE
Notice for licensing

### PR DESCRIPTION
**Pull request by dannify**
Add a NOTICE.txt file to keep a reference for the code we have used from different libraries.

### CODE REVIEW

File: NOTICE.txt [hunk 1]
```diff
@@ -0,0 +1,83 @@
+React-Spectrum
+=================
+The following is a list of sources from which code was used/modified in this codebase.
+
+-------------------------------------------------------------------------------
+This codebase contains a portion of code from react which can be obtained at:
+  * LICENSE:
```
Comment (devongovett) [inline NOTICE.txt:7]: should the link go first here? seems weird to say it "can be obtained at", and then no link.

File: packages/@react-aria/interactions/src/usePress.ts [hunk 2]
```diff
@@ -573,7 +573,10 @@ function isValidKeyboardEvent(event: KeyboardEvent): boolean {
   );
 }
 
-// Per: https://github.com/facebook/react/blob/3c713d513195a53788b3f8bb4b70279d68b15bcc/packages/react-interactions/events/src/dom/shared/index.js#L74-L87
```
Comment (devongovett) [inline packages/@react-aria/interactions/src/usePress.ts:576]: Might be nice to keep the links for easier reading?

File: packages/@react-aria/interactions/src/usePress.ts [hunk 3]
```diff
@@ -573,7 +573,10 @@ function isValidKeyboardEvent(event: KeyboardEvent): boolean {
...
+/*
```
Comment (devongovett) [inline packages/@react-aria/interactions/src/usePress.ts:576]: Does it need to be a block comment?

Comment (dannify) [reply packages/@react-aria/interactions/src/usePress.ts:576]: yep. I just thought it stood out more

**Pull request merged**